mlflow/store/tracking/file_store.py

```diff
@@ -2285,7 +2285,7 @@ def search_logged_models(
         self,
         experiment_ids: list[str],
         filter_string: Optional[str] = None,
-        datasets: Optional[list[str]] = None,
+        datasets: Optional[list[dict[str, Any]]] = None,
```

Comment:
can we use `TypedDict` for type hint since?

https://typing.python.org/en/latest/spec/typeddict.html#typed-dictionaries